### PR TITLE
Add Generate Models button to enable autogeneration of classes, associations, and mapping feature for relational databases

### PR DESCRIPTION
**Pull request by jake-kim1**
**Summary**

This PR adds a "Generate Models" button when right-clicking on a relational database, which calls the API for autogeneration of classes, and associations, and mapping. Additionally, when the user does not define any joins in the relational database, a pop-up appears with an option to either proceed or abort.

Note:
This PR depends on https://github.com/finos/legend-engine/pull/2070 which has been merged in Legend Engine

### CODE REVIEW

File: .changeset/sixty-carpets-cheat.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+'@finos/legend-application-studio': patch
+'@finos/legend-graph': patch
+---
+
+Add Generate Models button to enable autogeneration of classes, associations, and mapping feature for relational databases
```
Comment (MauricioUyaguari) [inline .changeset/sixty-carpets-cheat.md:6]: lets separate this to two changeste files

The above message can stay for `@finos/legend-application-studio` but should be a `minor` change. 
For the `@finos/legend-graph` we can keep empty.

**Review by MauricioUyaguari — CHANGES_REQUESTED** [review]
can we add a video of the workflow here. Not sure I agree with the approach of just generating the models directly without first showing the users the models that will be added to the project. this is the approach we do for generate database and we should follow a similar approach.

File: packages/legend-application-studio/src/components/editor/side-bar/Explorer.tsx [hunk 2]
```diff
@@ -523,6 +532,46 @@ const ExplorerContextMenu = observer(
         }
       },
     );
+    const generateModelsFromDatabaseSpecification =
+      editorStore.applicationStore.guardUnhandledError(async () => {
+        if (isRelationalDatabase(node?.packageableElement)) {
+          const databasePath: string = guaranteeNonEmptyString(
+            node?.packageableElement.path,
+          );
+          const graph: PureModel = editorStore.graphManagerState.graph;
```
Comment (YannanGao-gs) [inline packages/legend-application-studio/src/components/editor/side-bar/Explorer.tsx:541]: coudl we remove ': PureModel'?

File: .changeset/sixty-carpets-cheat.md (hunk 1 above)
Comment (MauricioUyaguari) [inline .changeset/sixty-carpets-cheat.md:6]:
````suggestion
---
'@finos/legend-application-studio': minor
'@finos/legend-graph': minor
---
Support generating models from relational database. 
````

File: packages/legend-application-studio/src/components/editor/side-bar/Explorer.tsx (hunk 2 above)
Comment (MauricioUyaguari) [inline packages/legend-application-studio/src/components/editor/side-bar/Explorer.tsx:541]:
````suggestion
          const graph = editorStore.graphManagerState.graph;
````

File: packages/legend-application-studio/src/components/editor/side-bar/Explorer.tsx [hunk 3]
```diff
@@ -523,6 +532,46 @@ const ExplorerContextMenu = observer(
         }
       },
     );
+    const generateModelsFromDatabaseSpecification =
+      editorStore.applicationStore.guardUnhandledError(async () => {
+        if (isRelationalDatabase(node?.packageableElement)) {
+          const databasePath: string = guaranteeNonEmptyString(
```
Comment (MauricioUyaguari) [inline packages/legend-application-studio/src/components/editor/side-bar/Explorer.tsx:538]:
````suggestion
          const databasePath = guaranteeNonEmptyString(
````

**Review by MauricioUyaguari — CHANGES_REQUESTED** [review]

File: packages/legend-application-studio/src/components/editor/side-bar/Explorer.tsx [hunk 4]
```diff
@@ -111,8 +111,14 @@ import {
   guaranteeRelationalDatabaseConnection,
   extractDependencyGACoordinateFromRootPackageName,
   type FunctionActivatorConfiguration,
+  Database,
+  type PureModel,
```
Comment (MauricioUyaguari) [inline packages/legend-application-studio/src/components/editor/side-bar/Explorer.tsx:115]:
````suggestion
 Database,
````

**Review by MauricioUyaguari — APPROVED** [review]

**Comment by jake-kim1** [comment]
https://github.com/finos/legend-studio/assets/60619826/4fdc50b1-c4b1-460d-a219-c511835024ef

**Pull request merged**